### DISCUSSION TITLE
Penggunaan library scorecard()

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#567] [Q&A] [answered]
Dear tim Algoritma,

Mohon sarannya untuk menentukan score min 300 dan max 850 pada syntax ini:
scorecards <- scorecard(bins,
                        logistic_step,
                        points0 = 500,
                        pdo = -30,
                        odds0 = 100,
                        basepoints_eq0 = TRUE)

Saya mendapatkan referensi syntax berikut namun hasilnya error
Tentukan batas skor minimum dan maksimum
score_min <- 300
score_max <- 850

Hitung nilai pdo dan basepoints
pdo <- -20
basepoints <- 600 - pdo * log(1/9)

Hitung nilai odds pada score maksimum
odds_max <- exp((score_max - basepoints) / pdo)

Hitung nilai odds pada score minimum
odds_min <- exp((score_min - basepoints) / pdo)

Tentukan nilai posprob_pop
posprob_pop <- 0.5 # atau nilai lain sesuai kebutuhan

Buat scorecard
sc <- scorecard(df, y = "target", x = c("x1", "x2", "x3", "x4"), 
                binning_params = list(cardinality_cutoff = 5),
                options = list(omit = NULL),
                positive = "1",
                score = list(pdo = pdo, basepoints = basepoints, odds.points = odds_max),
                smooth_params = list(smooth_method = "binsmooth", smooth_num = 4, smooth_criterion = "aic"),
                breaks_params = list(nbreaks = 6),
                monotone_constraints = NULL,
                max_score = score_max,
                min_score = score_min,
                posprob_pop = posprob_pop)

<img width="373" alt="image" src="https://user-images.githubusercontent.com/103427989/220866975-b1b1b382-701c-4fce-bce7-abd3488adce2.png">

Mohon sarannya,

Best regards,
JT

**Comment by finesaaa** [comment]
Dear @joejnt ,
Apakah boleh dilampirkan untuk link referensi lengkapnya pak? 
Apakah yang dimaksud library scorecard adalah ini https://github.com/ShichenXie/scorecard?
Jika iya, apakah bapak sudah melakukan instalasinya `install.packages("scorecard")`

**Comment by joejnt** [comment]
Dear @finesaaa 

Referensi yang saya gunakan adalah dari 
https://rpubs.com/harendra_16/Credit_Scoring_modelling dan
https://rpubs.com/ngyongkad/scorecard

Namun pada referensi tersebut tidak menyediakan informasi bagaimana setting parameter nilai minimum pada hasil score (pada case saya, nilai yang ingin ditampilkan adalah min 350 dan max 850)

Untuk library scorecard() sudah saya install.

Mohon saran dan petunjuknya.

Best regards,
JT

**Comment by idoali** [comment]
Selamat pagi pak. 

Keterangan error yang bapak dapatkan adalah : 

> *could not find function scorecard*

Ini artinya antara belum menginstall library `scorecard` atau belum di inisiasi diawal dengan cara `library(scorecard)`. Mungkin bisa dicek lagi pak

**Reply by joejnt** [reply]
Dear @idoali 

Untuk library scorecard() sudah saya install, tapi hasilnya seperti itu.

Best regards,
JT

**Comment by idoali** [comment, marked as answer]
Bisa juga dicoba untuk menggunakan fungsi scorecard dengan cara `scorecard::scorecard()` pak.
